django/contrib/admin/options.py

```diff
@@ -473,9 +473,12 @@ def to_field_allowed(self, request, to_field):
             for inline in admin.inlines:
                 registered_models.add(inline.model)
 
-        for related_object in (opts.get_all_related_objects(include_hidden=True) +
-                               opts.get_all_related_many_to_many_objects()):
-            related_model = related_object.model
+        related_objects = (
+            f for f in opts.get_fields(include_hidden=True)
+            if (f.auto_created and not f.concrete)
```

Comment:
On closer inspection - I think @PirosB3's version actually *might* be the simplest correct version here. The original code is looking for all related_objects and related_m2m_objects -- that's effectively what `auto_created` now covers - `auto_created` is being used in a lot of places `is_related_field` was being used previously. You can't *just* use `one_to_one` and `many_to_one`, because that misses `many_to_many`; you can't just use `not concrete`, because of `GenericRelation`. You have to use both `auto_created and not concrete` because of `AutoField`

We *could* put in a `many_to_many or many_to_one or one_to_one` clause for the sake of clarity, but that isn't strictly necessary based on the fields we have right now.